File: tree/is_full_bitree.py
# !/usr/bin/env python
# -*- coding: utf-8 -*-

class Node(object):
    def __init__(self, data, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right

def check_full(root):
    if root == None:
        return True
    if root.left == None and root.right != None:
        return False
    return check_full(root.left) and check_full(root.right)
    

tree = Node(1, Node(2, None, Node(5)), Node(3))
print(check_full(tree))

File: tree/is_full_bitree.cpp
#include <iostream>
using namespace std;

struct TreeNode{
    int val;
    TreeNode *left=nullptr;
    TreeNode *right=nullptr;
    TreeNode(int val):val(val){}
};

bool check(TreeNode *pRoot){
    if (pRoot == nullptr){
        return true;
    }
    if (pRoot->left == nullptr && pRoot->right != nullptr){
        return false;
    }
    return check(pRoot->left) && check(pRoot->right);
}

int main(void){
    TreeNode *root = new TreeNode(1);
    root->left = new TreeNode(2);
    cout<<check(root)<<endl;

    return 0;
}
